File: 프로그래머스/2/42578. 의상/의상.java
import java.util.*;

class Solution {
    public int solution(String[][] clothes) {
        int answer = 1;
        
        Map<String, Integer> map = new HashMap<>();
        
        for(int i=0; i<clothes.length; i++) {
            map.put(clothes[i][1], map.getOrDefault(clothes[i][1], 0)+1);
        }
                
        for(Integer n: map.values()) {
            answer *= (n+1);
        }
        
        return answer-1;
    }
}

File: 프로그래머스/2/42578. 의상/의상.py
def solution(clothes):
    answer = 1
    clothes_dict = {}
    
    for cloth in clothes:
        cloth_type = cloth[1]
        if cloth_type in clothes_dict:
            clothes_dict[cloth_type] += 1
        else:
            clothes_dict[cloth_type] = 2
    
    for v in clothes_dict.values():
        answer *= v
    
    return answer - 1